client/cli/src/runner.rs

```diff
@@ -177,45 +178,56 @@ impl<C: SubstrateCli> Runner<C> {
 
 	/// A helper function that runs a future with tokio and stops if the process receives the signal
 	/// `SIGTERM` or `SIGINT`.
-	pub fn run_subcommand<BU, B, BA, IQ, CL>(self, subcommand: &Subcommand, builder: BU)
-		-> Result<()>
+	pub fn run_subcommand<B, FBA, IQ, FCL, FB, LB, LCL, LBA>(
+		self, subcommand: &Subcommand, full_builder: FB, light_builder: LB,
+	) -> Result<()>
 	where
-		BU: FnOnce(Configuration)
-			-> sc_service::error::Result<(Arc<CL>, Arc<BA>, IQ, TaskManager)>,
+		FB: FnOnce(Configuration)
+			-> sc_service::error::Result<(Arc<FCL>, Arc<FBA>, IQ, TaskManager)>,
+		LB: FnOnce(Configuration)
+			-> sc_service::error::Result<(Arc<LCL>, Arc<LBA>, NetworkStatusSinks<B>, TaskManager)>,
 		B: BlockT + for<'de> serde::Deserialize<'de>,
-		BA: sc_client_api::backend::Backend<B> + 'static,
+		FBA: sc_client_api::backend::Backend<B> + 'static,
 		IQ: sc_service::ImportQueue<B> + 'static,
 		<B as BlockT>::Hash: FromStr,
 		<<B as BlockT>::Hash as FromStr>::Err: Debug,
 		<<<B as BlockT>::Header as HeaderT>::Number as FromStr>::Err: Debug,
-		CL: UsageProvider<B> + BlockBackend<B> + StorageProvider<B, BA> + Send + Sync +
+		FCL: UsageProvider<B> + BlockBackend<B> + StorageProvider<B, FBA> + Send + Sync +
 		'static,
+		LCL: HeaderBackend<B>,
+		LBA: sc_service::chain_ops::MaybeChtRootStorageProvider<B>,
 	{
 		let chain_spec = self.config.chain_spec.cloned_box();
 		let network_config = self.config.network.clone();
 		let db_config = self.config.database.clone();
 
 		match subcommand {
-			Subcommand::BuildSpec(cmd) => cmd.run(chain_spec, network_config),
+			Subcommand::BuildSpec(cmd) => {
+				let (client, backend, network_status_rx, task_manager) = light_builder(self.config)?;
```

Comment:
The `build-spec` command is now always executed with a light client. This is really really weird behavior...